bftengine/src/bcstatetransfer/BCStateTran.cpp

```diff
@@ -1481,8 +1481,12 @@ bool BCStateTran::onMessage(const AskForCheckpointSummariesMsg *m, uint32_t msgL
     if (!psd_->hasCheckpointDesc(i)) continue;
 
     DataStore::CheckpointDesc cpDesc = psd_->getCheckpointDesc(i);
-    std::unique_ptr<CheckpointSummaryMsg> msg =
-        std::unique_ptr<CheckpointSummaryMsg>(CheckpointSummaryMsg::create(cpDesc.rvbData.size()));
```

Comment:
@HristoStaykov and @glevkovich, would there be a safer way to allocate the `CheckpointSummaryMsg` so that we don't need a custom deleter? This approach seems very error-prone...